common/djangoapps/track/views/__init__.py

```diff
@@ -156,22 +152,14 @@ def task_track(request_info, task_info, event_type, event, page=None):
 
     # supplement event information with additional information
     # about the task in which it is running.
-    full_event = dict(event, **task_info)
-
-    # Get values from the task-level
-    # information, or just add placeholder values.
-    with eventtracker.get_tracker().context('edx.course.task', contexts.course_context_from_url(page)):
-        event = {
-            "username": request_info.get('username', 'unknown'),
```

Comment:
While the implementation of server_track looks like it will work correctly, using the same implementation for task_track will drop information.   The task_track call differs in that it relies on the request information from the original request to be passed to the task, and then the task_track calls were getting the request information explicitly from the request_info dict.

Reply:
From the changes, It seems that we are just not using the `host`, `IP` and `agent` information from the `request_info` dict. But after comparing the event logs from before and after this change, I can see that almost all of the information is present in the event, (just not in the context field after the change). The only missing field after the change is `client_id`.
Therefore, just to be on safe side, we can add the `request_info` dict in the event context. Would that be okay?

The track event seems larger since it has duplicate information.

Attaching the comparison SS
![image](https://user-images.githubusercontent.com/42166091/85658819-33b0d800-b6cd-11ea-8dc3-07d4f7c13df9.png)

Reply:
@brian @danialmalik made few changes to make task type event emitted via `eventtracking` compatible with `track` task event.
Here is the new comparison
![Screen Shot 2020-06-25 at 5 21 06 PM](https://user-images.githubusercontent.com/434995/85719127-92477780-b708-11ea-8c2d-a4857f1ef642.png)

Reply:
@brianhw this is now the task event looks like now after removing changes in the shim and using `contexts.course_context_from_url(page)` instead of `_get_course_context(page)` in `task_track` method. This would allow us to keep `course_id` and `org_id` keys in the context instead of removing them when they are empty.